### PR TITLE
Adds cash and coins

### PR DESCRIPTION
**Pull request by NoooneyDude**
Adds cash and coins. _Cash in hand, Mr. Rudd! Cash in hand._

### Purpose
- Adds cash.
- Adds coins.
    * Coins can be flipped.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/Dice/FlipCoin.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	{
+		netTransform.OnThrowStart.RemoveListener(ThrowStart);
+		netTransform.OnThrowEnd.RemoveListener(ThrowEnd);
+	}
+
+	#endregion Lifecycle
+
+	private void UpdateSprite()
+	{
+		spriteHandler.ChangeSprite(0);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Dice/FlipCoin.cs]: Do constants for sprites indexes

File: UnityProject/Assets/Scripts/Items/Dice/RollDie.cs [hunk 2]
```diff
@@ -69,7 +69,11 @@ private void OnDisable()
 	{
 		netTransform.OnThrowStart.RemoveListener(ThrowStart);
 		netTransform.OnThrowEnd.RemoveListener(ThrowEnd);
-		cookable.OnCooked.RemoveListener(Cook);
+
+		if (cookable != null)
+		{
+			cookable.OnCooked.RemoveListener(Cook);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Dice/RollDie.cs]: UnityEvents are cool when you know you have many listeners, for 1 listener it is better to use native delegates.

Here is some data https://jacksondunstan.com/articles/3335

Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/Dice/RollDie.cs]: but if you are exposing the event to inspector, then Unityevent is the way to go.

File: UnityProject/Assets/Scripts/Items/Dice/FlipCoin.cs [hunk 3]
```diff
@@ -88,9 +93,9 @@ private void ThrowEnd(ThrowInfo throwInfo)
 
 	private IEnumerator Flip()
 	{
-		// Sprite SO 1 is a flip animation.
+		// The flip animation sprite SO has only only one variant.
 		spriteHandler.ChangeSpriteVariant(0);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Dice/FlipCoin.cs:97]: only missed this one

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Items/Dice/FlipCoin.cs:97]: No, I deliberately left that one to zero, as the flip animation doesn't have any other variants.

Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/Dice/FlipCoin.cs:97]: oh alright. Once tests are done, I'm merging.

**Pull request merged**